### PR TITLE
Use ReadableStream

### PR DESCRIPTION
**Pull request by MaxLeiter**
Fixes support for Cloudflare workers and vercel edge runtime. ReadableStreams are available in Node.js since Node 18: https://nodejs.org/docs/latest-v18.x/api/webstreams.html#web-streams-api

### CODE REVIEW

**Review by zeke — APPROVED** [review]
I don't know much about the internal of this stuff, but this look sensible to me, and the tests are passing. I'm cool with shipping this, barring any opposition from @mattt 

@MaxLeiter your PR body explains the purpose of this change, but can you add a bit more detail that I can use for some release notes about this change?

File: lib/stream.js [hunk 1]
```diff
@@ -41,95 +30,82 @@ class ServerSentEvent {
...
+        case 'id':
+          id = value.trim();
+          break;
+        case 'retry':
+          retry = parseInt(value.trim(), 10);
+          break;
       }
     }
+    return new ServerSentEvent(event, data.trim(), id, retry);
   }
```
Comment (mattt) [inline lib/stream.js:109]: I think `Stream` needs to define an async iterator to keep compatibility with its usage in the `stream` method.


```suggestion
  }
  
  [Symbol.asyncIterator]() {
    const reader = this.readableStream.getReader();
    return {
      next: async () => {
        const { done, value } = await reader.read();
        return { done, value };
      },
      return: () => {
        reader.releaseLock();
        return { done: true };
      },
      throw: (error) => {
        reader.releaseLock();
        throw error;
      },
    };
  }
```

**Review by mattt — COMMENTED** [review]
Thanks so much for taking a look at this, @MaxLeiter. Overall, this is looking much nicer! I have one refactoring suggestion to move `parseEvent` to a static method on the SSE class (patch here: [0001.patch](https://github.com/replicate/replicate-javascript/files/14202090/0001-Refactor-SSE-construction-to-static-method-on-Server.patch))

The only blocker is that I think we need to define the async iterator for the class if we're keeping the existing usage in `replicate.stream`. Left a suggestion in a comment with some untested code.

Speaking of untested, I don't believe the `stream` method has any test coverage at the moment, because I couldn't figure out how to get SSE to play nicely with nock. I tested this locally by hand for the initial version, but it'd be great to automate this.

**Comment by mattt** [comment]
Hi @MaxLeiter. Thanks again for your help with this. We just merged #214 and released [v0.28.0](https://github.com/replicate/replicate-javascript/releases/tag/v0.28.0), which should resolve any issues you were having. We also added a Cloudflare Workers integration test suite with #217. If you have a chance, please give that a try and confirm that's working for you 🙇

**Pull request closed without merging**